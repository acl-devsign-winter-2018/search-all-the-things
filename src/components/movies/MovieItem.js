import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Movie.css';

export default class MovieItem extends Component {

  static propTypes = {
    Title: PropTypes.string.isRequired,
    Rated: PropTypes.string.isRequired,
    Released: PropTypes.string.isRequired,
    Runtime: PropTypes.string.isRequired,
    Genre: PropTypes.string.isRequired,
    Director: PropTypes.string.isRequired,
    Writer: PropTypes.string.isRequired,
    Actors: PropTypes.string.isRequired,
    Poster: PropTypes.string.isRequired,
    imdbRating: PropTypes.string.isRequired
  };

  render() {
    const { Title, Rated, Released, Runtime, Genre, Director, Writer, Actors, Poster, imdbRating } = this.props;

    return (
      <div className="movie-item">
       
        <div className="poster">
          {(Poster !== 'N/A') && <img src={Poster}/>}
          {(Poster === 'N/A') && 
            <div className="missing">
              <span className="missing-fill">:(</span>
            </div>
          }
        </div>

        <div className="movie-description">
          <h3>{Title}</h3>
          <p>Rated: {Rated}</p>
          <p>Film Released: {Released}</p>
          <p>Runtime: {Runtime}</p>
          <p>Genre: {Genre}</p>
          <p>Director: {Director}</p>
          <p>Writers: {Writer}</p>
          <p>Actors: {Actors}</p>
          <p>IMDB Rating: {imdbRating}</p>
        </div>
      </div>
    );
  }
}