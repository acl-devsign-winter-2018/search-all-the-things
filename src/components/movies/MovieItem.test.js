import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import MovieItem from './MovieItem';

describe('Movie Item', () => {
  it('renders movie item', () => {
    const Image = 'https://images-na.ssl-images-amazon.com/images/M/MV5BYzc4ODgyZmYtMGFkZC00NGQyLWJiMDItMmFmNjJiZjcxYzVmXkEyXkFqcGdeQXVyNDYyMDk5MTU@._V1_SX300.jpg';
    const wrapper = shallow(
      <MovieItem 
        Poster={Image}
        Title="Batman Begins"
        Rated="PG-13"
        Released="15 Jun 2005"
        Runtime="140 min"
        Genre="Action, Adventure, Thriller"
        Director="Christopher Nolan"
        Writer="Bob Kane (characters), David S. Goyer (story), Christopher Nolan (screenplay), David S. Goyer (screenplay)"
        Actors="Christian Bale, Michael Caine, Liam Neeson, Katie Holmes"
        imdbRating="8.3"
      />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});