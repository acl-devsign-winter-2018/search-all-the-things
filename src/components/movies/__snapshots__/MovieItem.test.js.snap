// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Item renders movie item 1`] = `
<div
  className="movie-item"
>
  <div
    className="poster"
  >
    <img
      src="https://images-na.ssl-images-amazon.com/images/M/MV5BYzc4ODgyZmYtMGFkZC00NGQyLWJiMDItMmFmNjJiZjcxYzVmXkEyXkFqcGdeQXVyNDYyMDk5MTU@._V1_SX300.jpg"
    />
  </div>
  <div
    className="movie-description"
  >
    <h3>
      Batman Begins
    </h3>
    <p>
      Rated: 
      PG-13
    </p>
    <p>
      Film Released: 
      15 Jun 2005
    </p>
    <p>
      Runtime: 
      140 min
    </p>
    <p>
      Genre: 
      Action, Adventure, Thriller
    </p>
    <p>
      Director: 
      Christopher Nolan
    </p>
    <p>
      Writers: 
      Bob Kane (characters), David S. Goyer (story), Christopher Nolan (screenplay), David S. Goyer (screenplay)
    </p>
    <p>
      Actors: 
      Christian Bale, Michael Caine, Liam Neeson, Katie Holmes
    </p>
    <p>
      IMDB Rating: 
      8.3
    </p>
  </div>
</div>
`;
